packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.vue

```diff
@@ -57,18 +56,13 @@ const isExpanded = useVModel({
   key: "open",
   default: false,
 });
-
-const triggerBindings = computed(() => ({
-  onClick: () => (isExpanded.value = !isExpanded.value),
-  "aria-expanded": isExpanded.value,
-  "aria-haspopup": "dialog",
-}));
 </script>
 
 <template>
   <OnyxBasicPopover :class="['onyx-dialog', densityClass]" v-bind="props" :open="isExpanded">
-    <template #default>
-      <slot name="trigger" :trigger="triggerBindings"> </slot>
+    <template #default="{ trigger }">
+      <slot name="trigger" :trigger="{ ...trigger, onClick: () => (isExpanded = !isExpanded) }">
```

Comment:
The click should not be needed if we use the "@update:open" event on the basic popover :)